src/TestFramework/Coverage/JUnit/JUnitReportLocator.php

```diff
@@ -0,0 +1,122 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\TestFramework\Coverage\JUnit;
+
+use function array_map;
+use function count;
+use function current;
+use function file_exists;
+use function implode;
+use Infection\FileSystem\Locator\FileNotFound;
+use function iterator_to_array;
+use function Safe\sprintf;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+use Webmozart\PathUtil\Path;
+
+/**
+ * @internal
+ * @final
+ */
+class JUnitReportLocator
+{
+    private $coveragePath;
+    private $defaultJUnitPath;
+
+    /**
+     * @var string|null
+     */
+    private $jUnitPath;
+
+    public function __construct(string $coveragePath, string $defaultJUnitPath)
+    {
+        $this->coveragePath = Path::canonicalize($coveragePath . '/..');
+        $this->defaultJUnitPath = Path::canonicalize($defaultJUnitPath);
+    }
+
+    /**
+     * @throws FileNotFound
+     */
+    public function locate(): string
+    {
+        if ($this->jUnitPath !== null) {
+            return $this->jUnitPath;
+        }
+
+        // This is the JUnit path enforced before. It is also the one recommended by the
+        // CoverageChecker hence it makes sense to try this one first before attempting any more
+        // expensive lookup
+        if (file_exists($this->defaultJUnitPath)) {
+            return $this->jUnitPath ?? $this->defaultJUnitPath;
+        }
+
+        $files = iterator_to_array(
+            Finder::create()
+                ->files()
+                ->in($this->coveragePath)
+                ->name('/^(.+\.)?junit\.xml$/i')
+                ->sortByName(),
+            false
+        );
+
+        if (count($files) > 1) {
+            throw new FileNotFound(sprintf(
+                'Could not locate the JUnit file: more than one file has been found with the'
+                . ' pattern "*.junit.xml": "%s"',
+                implode(
+                    '", "',
+                    array_map(
+                        static function (SplFileInfo $fileInfo): string {
+                            return Path::canonicalize($fileInfo->getPathname());
+                        },
+                        $files
+                    )
+                )
+            ));
+        }
+
+        $junitFileInfo = current($files);
+
+        if ($junitFileInfo !== false) {
+            return $this->jUnitPath ?? Path::canonicalize($junitFileInfo->getPathname());
```

Comment:
Same question: do we need to check $this->jUnitPath here? If it's not `null`, we will early return on line 76 I guess

Reply:
Yes simple `=` is more appropriate here!